jwk_client.go

```diff
@@ -67,20 +66,20 @@ func (j *JWKClient) GetKey(ID string) (jose.JSONWebKey, error) {
 	j.mu.Lock()
```

Comment:
This is interesting, as it serves 3 purposes now. To serializes `GetKey` calls AND it protect accessed to the maps AND it only allows one download keys at a time. I wonder if we could do better for performance.